src/site/content/en/learn/css/specificity/index.md

```diff
@@ -0,0 +1,363 @@
+
+### ID selector
+
+An ID selector gets **100 points of specificity**,
+as long as you use an ID selector (`#myID`) and not an attribute selector (`[id="myID"]`).
+
+```css
+#myID {
+	color: red
+}
+```
+
+### Inline style attribute
+
+CSS applied directly to the `style` attribute of the HTML element,
+gets a **specificity score of 1,000 points**.
+This means that in order to override it in CSS,
+you have to write an extremely specific selector.
+
+```html
+<div style="color: red"></div>
+```
+
+### `!important` rule
+
+Lastly, an `!important` at the end of a CSS value gets a specificity score of **10,000 points**.
+This is the highest specificity that one individual item can get.
+
+An `!important` rule is applied to a CSS property,
+so everything in the overall rule (selector and properties) does not get the same specificity score.
+
+```css
+.my-class {
+	color: red !important; /* 10,000 points */
+	background: white; /* 10 points */
+}
+```
+
+@@TODO: self-assessment 1
+
+## Specificity in context
+
+The specificity of each selector that matches an element is added together.
+Consider this example HTML:
+
+```html
+<a class="my-class another-class" href="#">A link</a>
+```
+
+This link has two classes on it.
+Add the following CSS, and it gets **1 point of specificity**:
+
+```css
+a {
+  color: red;
+}
+```
+
+Reference one of the classes in this rule,
+it now has **11 points of specificity**:
+
+```css
+a.my-class {
+  color: green;
+}
+```
+
+Add the other class to the selector,
+it now has  **21 points of specificity**:
+
+```css
+a.my-class.another-class {
+  color: rebeccapurple;
+}
+```
+
+Add the `href` attribute to the selector,
+it now has **31 points of specificity**:
+
+```css
+a.my-class.another-class[href] {
+  color: goldenrod;
+}
+```
+
+Finally,add a `:hover` pseudo-class to all of that,
+the selector ends up with **41 points of specificity**:
+
+```css
+a.my-class.another-class[href]:hover {
+  color: lightgrey;
+}
+```
+
+@@TODO: self-assessment 2
+
+## Visualizing specificity
+
+In diagrams and specificity calculators,
+the specificity is often visualized like this:
+
+{% Img src="image/VbAJIREinuYvovrBzzvEyZOpw5w1/d1q5VKbFOQG2TBXoBtgm.png", alt="A diagram demonstrating most specific to least specific selectors", width="800", height="474" %}
+
+The left group is `id` selectors.
+The second group is class, attribute, and pseudo-class selectors.
+The final group is element and pseudo-element selectors.
+
+For reference, the following selector is `0-4-1`:
+
+```css
+a.my-class.another-class[href]:hover {
+  color: lightgrey;
+}
+```
+
+@@TODO: self-assessment 3
+
+## Pragmatically increasing specificity
+
+Let's say we have some CSS that looks like this:
+
+```css
+.my-button {
+	background: blue;
+}
+
+button[onclick] {
+	background: grey;
+}
+```
+
+With HTML that looks like this:
+
+```html
+<button class="my-button" onclick="alert('hello')">Click me</button>
+```
+
+<figure class="w-figure">
+{% Codepen {
```

Comment:
@robdodson @devnook would it be possible to have an option for which tabs to show? I.e. I think here, showing the CSS tab next to the result is useful. Current default just shows the default.

Reply:
I believe that option exists ([handbook](https://web.dev/handbook/web-dev-components/#codepen)). To show CSS and results, you'd do the following:

```njk
{% Codepen {
  user: 'robdodson',
  id: 'GRroyyX',
  height: 300,
  theme: 'dark',
  tab: 'css,result',
  allow: ['geolocation']
} %}
```

Reply:
Maybe this is a task we can do as a preflight, go through them all and make sure they each show the optimal arrangement of tabs?

Reply:
Sounds good, i'll make an issue to track this

Reply:
#5081